src/test/java/com/sgordievskiy/spring/controller/TodoControllerIT.java

```diff
@@ -0,0 +1,247 @@
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class TodoControllerIT {
+
+  protected static MockMvc mockMvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @BeforeAll
+  static void beforeAll(
+    @Autowired DataSource dataSource,
+    @Autowired WebApplicationContext applicationContext
+  ) throws SQLException {
+    mockMvc = MockMvcBuilders
+      .webAppContextSetup(applicationContext)
+      .build();
+    teardown(dataSource);
+    try (Connection connection = dataSource.getConnection()) {
+      connection.setAutoCommit(true);
+      ScriptUtils.executeSqlScript(
+        connection,
+        new ClassPathResource("database/add-default-task.sql")
+      );
+    }
+  }
+
+  @AfterAll
+  static void afterAll(@Autowired DataSource dataSource) {
+    teardown(dataSource);
+  }
+
+  @SneakyThrows
+  static void teardown(DataSource dataSource) {
+    try (Connection connection = dataSource.getConnection()) {
+      connection.setAutoCommit(true);
+      ScriptUtils.executeSqlScript(
+        connection,
+        new ClassPathResource("database/clear-db.sql")
+      );
+    }
+  }
+
+  @Test
+  @DisplayName("Create a new task")
+  @Sql(
+    scripts = "classpath:database/delete-test-todo.sql",
+    executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+  )
+  void createTodo_Success() throws Exception {
+    LocalDateTime date = LocalDateTime.of(2024, 11, 7, 12, 15, 30);
+    TodoCreateDto requestDto = new TodoCreateDto(
+      "Title",
+      "Description",
+      date,
+      null
+    );
+
+    TodoResponseDto expected = new TodoResponseDto(
+      1L,
+      "Title",
+      "Description",
+      date,
+      Priority.MEDIUM,
+      Status.PENDING,
+      date,
+      date,
+      1L
+    );
+
+    String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+    MvcResult result = mockMvc.perform(
+        post("/todos")
+          .content(jsonRequest)
+          .contentType(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(status().isOk())
+      .andReturn();
+
+    TodoResponseDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), TodoResponseDto.class);
+    Assertions.assertNotNull(actual);
+    Assertions.assertNotNull(actual.getId());
+    Assertions.assertEquals(expected.getTitle(), actual.getTitle());
+    Assertions.assertEquals(expected.getDescription(), actual.getDescription());
+    Assertions.assertEquals(expected.getDueDate(), actual.getDueDate());
+    Assertions.assertEquals(expected.getPriority(), actual.getPriority());
+    Assertions.assertEquals(expected.getStatus(), actual.getStatus());
+  }
+
+  @Test
+  @DisplayName("Update existing task")
+  @Sql(
+    scripts = {"classpath:database/clear-db.sql", "classpath:database/add-default-task.sql"},
```

Comment:
add-default-task.sql - not the best approach.

You have several options how to insert data:
- using `@BeforeAll` (already applied)
- using `@BeforeEach`
- using `@Sql` with Sql.ExecutionPhase.BEFORE_TEST_METHOD

Consider using these options to insert necessary data and corresponding methods-antagonists to remove inserted data.